File: src/main/java/software/aws/glue/tableversions/utils/SQSUtil.java
//Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
//SPDX-License-Identifier: MIT-0

package software.aws.glue.tableversions.utils;

import java.util.HashMap;
import java.util.Map;

import com.amazonaws.services.sqs.AmazonSQS;
import com.amazonaws.services.sqs.model.MessageAttributeValue;
import com.amazonaws.services.sqs.model.SendMessageRequest;
import com.amazonaws.services.sqs.model.SendMessageResult;

public class SQSUtil {

	/**
	 * This method send a message to SQS queue.
	 * @param sqs
	 * @param queueURI
	 * @param message
	 * @param executionBatchId
	 * @param databaseName
	 * @return
	 */
	public boolean sendTableSchemaToSQSQueue(AmazonSQS sqs, String queueURI, String message, long executionBatchId,
			String databaseName) {
		int statusCode = 400;
		boolean messageSentToSQS = false;
		Map<String, MessageAttributeValue> messageAttributes = new HashMap<>();
		messageAttributes.put("ExecutionBatchId", new MessageAttributeValue().withDataType("String.ExecutionBatchId")
				.withStringValue(Long.toString(executionBatchId)));
		SendMessageRequest req = new SendMessageRequest().withQueueUrl(queueURI).withMessageBody(message)
				.withMessageGroupId(databaseName).withMessageAttributes(messageAttributes);
		try {
			SendMessageResult sendMsgRes = sqs.sendMessage(req);
			statusCode = sendMsgRes.getSdkHttpMetadata().getHttpStatusCode();
		} catch (Exception e) {
			e.printStackTrace();
			System.out.println("Exception thrown while writing message to SQS. " + e.getLocalizedMessage());
		}
		if (statusCode == 200) {
			messageSentToSQS = true;
		} else
			System.out.printf("Cannot write Table schema %s to SQS queue. \n", message);
		return messageSentToSQS;
	}

}

File: src/utils/sqs_util.py
# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: MIT-0

import boto3
import os


def get_sqs_client():
    """
    SQS Client

    :return:
        """
    return boto3.client('sqs')


def send_message(client, queue_url, message, message_attribute_key, message_attribute_value, group_id):
    """
    Sends message to SQS

    :param client:
    :param queue_url:
    :param message:
    :param message_attribute_key:
    :param message_attribute_value:
    :param group_id:
    :return message_sent:
    """
    message_sent = False
    response = client.send_message(
        QueueUrl=queue_url,
        MessageBody=message,
        MessageAttributes={
            message_attribute_key: {
                'StringValue': str(message_attribute_value),
                'DataType': 'String'
            }
        },
        MessageGroupId=group_id
    )
    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
        message_sent = True
    return message_sent
